core/bin/external_node/src/main.rs

```diff
@@ -460,6 +630,59 @@ struct Cli {
     /// This is an experimental and incomplete feature; do not use unless you know what you're doing.
     #[arg(long)]
     enable_snapshots_recovery: bool,
+    /// Comma-separated list of components to launch.
+    #[arg(long, default_value = "all")]
+    components: ComponentsToRun,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Component {
+    HttpApi,
+    WsApi,
+    Tree,
+    TreeApi,
+    Core,
+}
+
+#[derive(Debug)]
+pub struct Components(pub Vec<Component>);
+
+impl FromStr for Components {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "api" => Ok(Components(vec![Component::HttpApi, Component::WsApi])),
+            "http_api" => Ok(Components(vec![Component::HttpApi])),
+            "ws_api" => Ok(Components(vec![Component::WsApi])),
+            "tree" => Ok(Components(vec![Component::Tree])),
+            "tree_api" => Ok(Components(vec![Component::TreeApi])),
+            "core" => Ok(Components(vec![Component::Core])),
+            "all" => Ok(Components(vec![
+                Component::HttpApi,
+                Component::WsApi,
+                Component::Tree,
+                Component::Core,
+            ])),
+            other => Err(format!("{} is not a valid component name", other)),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ComponentsToRun(Vec<Component>);
```

Comment:
Bikeshedding: IMO, two-layer parsing looks confusing. IMO, it'd be better to have a single `Components` wrapper and refactor the inner parsing layer (i.e., one that happens after the input is split by `,`) as an function or an inherent static method in `Components`.